tiledb/sm/filesystem/vfs.cc

```diff
@@ -58,12 +59,14 @@ namespace tiledb::sm {
 
 VFS::VFS(
     stats::Stats* const parent_stats,
+    Logger* const logger,
```

Comment:
I wonder if we should just have the VFS take a context to simplify the parameter list.